ddtrace/tracer/handler.go

```diff
@@ -0,0 +1,235 @@
+		return
+	}
+	h.wg.Add(1)
+	h.climit <- struct{}{}
+	go func(p *payload) {
+		defer func(start time.Time) {
+			<-h.climit
+			h.wg.Done()
+			h.config.statsd.Timing("datadog.tracer.flush_duration", time.Since(start), nil, 1)
+		}(time.Now())
+		size, count := p.size(), p.itemCount()
+		log.Debug("Sending payload: size: %d traces: %d\n", size, count)
+		rc, err := h.config.transport.send(p)
+		if err != nil {
+			h.config.statsd.Count("datadog.tracer.traces_dropped", int64(count), []string{"reason:send_failed"}, 1)
+			log.Error("lost %d traces: %v", count, err)
+		} else {
+			h.config.statsd.Count("datadog.tracer.flush_bytes", int64(size), nil, 1)
+			h.config.statsd.Count("datadog.tracer.flush_traces", int64(count), nil, 1)
+			if err := h.prioritySampling.readRatesJSON(rc); err != nil {
+				h.config.statsd.Incr("datadog.tracer.decode_error", nil, 1)
+			}
+		}
+	}(h.payload)
+	h.payload = newPayload()
+}
+
+type hexInt uint64
+
+func (i hexInt) MarshalJSON() ([]byte, error) {
+	return []byte(`"` + strconv.FormatUint(uint64(i), 16) + `"`), nil
+}
+
+type jsonSpan struct {
+	TraceID  hexInt             `json:"trace_id"`
+	SpanID   hexInt             `json:"span_id"`
+	ParentID hexInt             `json:"parent_id"`
+	Name     string             `json:"name"`
+	Resource string             `json:"resource"`
+	Error    int32              `json:"error"`
+	Meta     map[string]string  `json:"meta"`
+	Metrics  map[string]float64 `json:"metrics"`
+	Start    int64              `json:"start"`
+	Duration int64              `json:"duration"`
+	Service  string             `json:"service"`
+}
+
+type lambdaTraceWriter struct {
+	config    *config
+	payload   bytes.Buffer
+	hasTraces bool
+}
+
+var _ traceWriter = &lambdaTraceWriter{}
+
+func newLambdaTraceWriter(c *config) *lambdaTraceWriter {
+	w := &lambdaTraceWriter{
+		config: c,
+	}
+	w.resetPayload()
+	return w
+}
+
+const payloadLimit = 256 * 1024 // log line limit for cloudwatch
+
+func (h *lambdaTraceWriter) resetPayload() {
+	h.payload.Reset()
+	h.payload.WriteString(`{"traces": [`)
+	h.hasTraces = false
+}
+
+func (h *lambdaTraceWriter) addSpan(s *span) error {
+	js := jsonSpan{
+		TraceID:  hexInt(s.TraceID),
+		SpanID:   hexInt(s.SpanID),
+		ParentID: hexInt(s.ParentID),
+		Name:     s.Name,
+		Resource: s.Resource,
+		Error:    s.Error,
+		Meta:     s.Meta,
+		Metrics:  s.Metrics,
+		Start:    s.Start,
+		Duration: s.Duration,
+		Service:  s.Service,
+	}
+	e := json.NewEncoder(&h.payload)
+	err := e.Encode(js)
+	if err != nil {
+		return err
+	}
+	// cut trailing newline
+	h.payload.Truncate(h.payload.Len() - 1)
+	return nil
+}
+
+type encodingError struct {
+	cause      error
+	dropReason string
+}
+
+func (h *lambdaTraceWriter) addTrace(trace []*span) (int, *encodingError) {
+	startLen := h.payload.Len()
+	if !h.hasTraces {
+		h.payload.WriteByte('[')
+	} else {
+		h.payload.WriteString(", [")
+	}
+	written := 0
+	for i, s := range trace {
+		l := h.payload.Len()
```

Comment:
```suggestion
		n := h.payload.Len()
```
Does `n` work? Seems more common